src/components/pages/Users/UserPanel/Fields/Fields.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { Formik } from "formik";
+import { vi } from "vitest";
+
+import { renderComponent } from "test/utils";
+
+import Fields from "./Fields";
+import { Label, type FormFields } from "./types";
+
+test("should show the correct fields", () => {
+  renderComponent(
+    <Formik<FormFields>
+      initialValues={{
+        email: "test@email.com",
+        firstName: "First",
+        lastName: "Last",
+      }}
+      onSubmit={vi.fn()}
+    >
+      <Fields setIsDirty={vi.fn()} />
+    </Formik>,
+  );
+  expect(screen.getByRole("textbox", { name: Label.EMAIL })).toHaveValue(
+    "test@email.com",
+  );
+  expect(screen.getByRole("textbox", { name: Label.FIRST_NAME })).toHaveValue(
+    "First",
+  );
+  expect(screen.getByRole("textbox", { name: Label.LAST_NAME })).toHaveValue(
+    "Last",
+  );
+});
+
+test("should set isDirty to true when fields are modified", async () => {
+  const mockIsDirty = vi.fn();
+  renderComponent(
+    <Formik<FormFields> initialValues={{ email: "" }} onSubmit={vi.fn()}>
+      <Fields setIsDirty={mockIsDirty} />
+    </Formik>,
+  );
+  expect(mockIsDirty).toHaveBeenCalledWith(false);
```

Comment:
This needs to be `toHaveBeenLastCalledWith` otherwise it will check if it was EVER called with `false`.